cubeviz/tests/test_image_viewer.py

```diff
@@ -44,3 +58,48 @@ def test_stats_box_with_subset(cubeviz_layout):
     results_string = r"min={:.4}, max={:.4}, median={:.4}, μ={:.4}, σ={:.4}".format(*results)
 
     assert results_string == cl_viewer.parent().stats_text.text()
+
+    # Remove the ROI/subset
+    dc = cubeviz_layout.session.application.data_collection
+    dc.remove_subset_group(dc.subset_groups[0])
+
+
+def test_overlay(moment_maps_gui, cubeviz_layout):
+    # Only "No Overlay" option available in combobox
+    assert cubeviz_layout._overlay_controller._overlay_image_combo.count() == 1
+
+    # Create moment map GUI
+    mm = moment_maps_gui
+    mm.display()
+    mm.order_combobox.setCurrentIndex(0)
+    mm.data_combobox.setCurrentIndex(0)
+
+    # Call calculate function and get result
+    mm.calculateButton.click()
+
+    # Second option in combobox for moment map overlay
+    assert cubeviz_layout._overlay_controller._overlay_image_combo.count() == 2
+
+    cl_viewer = cubeviz_layout.split_views[1]._widget
+    assert len(cl_viewer.axes.images) == 1
+
+    # Set overlay and check changing of settings works
+    cubeviz_layout._overlay_controller._overlay_image_combo.setCurrentIndex(1)
+    cubeviz_layout._overlay_controller._overlay_colormap_combo.setCurrentIndex(10)
+    cubeviz_layout._overlay_controller._alpha_slider.setValue(50)
+
+    assert len(cl_viewer.axes.images) == 2
+
+    # Return to "No Overlay"
+    cubeviz_layout._overlay_controller._overlay_image_combo.setCurrentIndex(0)
+
+    assert len(cl_viewer.axes.images) == 1
+
+    # Remove the moment map data set
```

Comment:
It actually might be a good idea to move all of this logic to the fixture that resets app state after each test.